### ISSUE TITLE
Mintty issue? Problem could be me...

### ISSUE THREAD

**Issue by ludasmokva** [oneclick/rubyinstaller2#78]
This project is for Ruby version 2.4 and newer.
For Ruby versions < 2.4 please file an issue [here]( https://github.com/oneclick/rubyinstaller/issues/new ).

## What problems are you experiencing? 
Not sure...yet. First time user so getting stuck on first stage. This could be simply my complete and utter ignorance. And if it is, very sorry. But could not find any info on it.  
Installed latest version 2.4.1-2 x64 so I can install Sass. While install went through (seemed ok), the Minnty window popped up. No clue what to do with it. Tried typing something next to $, no text appears. Message I get upon enter: 
$ 602992160 [main] bash 5624 child_info::sync: wait failed, pid 2012, Win32 error 0
602993446 [main] bash 5624 fork: child -1 - forked process 2012 died unexpectedly, retry 0, exit code 0x1, errno 11
bash: fork: retry: No child processes
bash: cut: command not found
 
This is the only thing that keeps repeating. Like I said, curser is flashing but cannot type anything. 

## Steps to reproduce:  
It was just after install. Simply started auto install, then pressed 'Enter' when it suggested to do so if I don't know what I'm doing :) - it was something like ENTER [1,2,3] but I just pressed enter.  The process continued and then the Mintty window popped up and the rest is explained above. 

## What's the output from `ridk version`? 
Don't understand this. Sorry. 

OTHER: Not sure if this is an issue or not. Sass appears to have installed ok but I haven't tried to use it yet. I truly do not understand the purpose of Mintty (yes.. I read their wiki - trying to translate into english :) ). But it seems to me that it is a form of CMD and as such, I should be able to  type commands into it. I cannot.

A response in plain english would be really really appreciated. I would hate to waste your time by asking 'what does that mean?' over and over again. 
Thx.

**Comment by larskanis** [comment]
This is very likely an issue with some kind of anti virus software. The AV "memory protection" or "realtime protection" features sometimes interferes with Cygwin/MSYS2 fork implementation. There was a similar issue #72 lately. Cygwin maintains [a list of such applications](https://cygwin.com/faq/faq.html#faq.using.bloda). Can you please check whether you're using one of these products and reinstall RubyInstaller with it disabled?

**Comment by ludasmokva** [comment]
Ok. I reviewed and double checked the list you linked to (thank you for the btw.)
I doubted it was an antivirus as I don't have one installed. I use Malwarebytes, Defender and common sense. :) I did double check if Defender caught or reacted to anything, and it didn't.
I did have my Firewall pop-up during install (Windows Firewall with Glasswire), but I gave it permission to allow the app. Still, I double checked, and it is not blocking anything that I haven't specifically told it to; Ruby, gpkeys and pacman (all related to this install) seem to have access.  

To be honest, I really don't know what to do with that mintty window. Should I close it? Should I uninstall it? I can't type anything into it. Pressing enter simply gives me either a blank line or something like the above or:
$ bash: eisalssgmgmkd3ep-lbb: command not found
or
$ bash: vv: command not found
or
$ bash: -e: command not found

What's funny is that I have no idea who is giving it these commands.
Do I need this?

**Comment by larskanis** [comment]
The MSYS2 installer opens a mintty window on it's last page. You can ignore and close this window, it isn't needed for Ruby.

However the error message
```child_info::sync: wait failed [...] fork: child -1 - forked process 2012 died unexpectedly```
indicates, that something goes wrong on a very low level within the fork implementation of MSYS2/Cygwin. Ruby will probably work and some tools of MSYS2 as well, but other tools, like pacman will probably fail like the bash program in mintty.

**Comment by ludasmokva** [comment]
Ok. 
I don't see pacman in processes, so maybe you are right... or maybe it shouldn't be appearing there. I am really unfamiliar with this program. I simply wanted to write some code in scss but am thinking I may be much better off just sticking with standard css . :) From what you said, its quite possible that even Ruby won't function properly and I have no way of figuring out if that is the case or not.
Can I send you any other information that would help you in identifying the problem? A list of programs? Settings? A report? (For more personal/security info, please provide a more secure way for me to provide the info. )
I am running win7 ultimate 64 if that helps in any way.

**Comment by ludasmokva** [comment]
Just an update...
I installed the new  2.4.2-2 and everything seemed to work out fine. No error messages.
And you were correct about pacman... the new install recognized that MSYS2 was already installed, but recognized that pacman wasn't. Seemed to attempt to rectify that, then said a version of 'success' and finished. No error messages popped up, and the Mintty window didn't show up. 
So the new version seemed to fix whatever went wrong the first time. Yaay! 

I do have a  quick question... I assume that Ruby installed over the previous files, although there was no prompt to say that. 
In the Explorer window, the folder is simply c: Ruby24-x64.  
But when I check out the folders through the Start prompt, it shows it as 2.4.1-2x64 which would seem to indicate that the new version did not install. 
But within the folder it gives me an option to uninstall 2.4.1 _and_ 2.4.2 which I guess means that the new version did in fact install in the same folder. Kind of confusing if you're trying to figure out what's in your system... and especially confusing as my uninstaller only shows the 2.4.2 version.  So I'm kind of wondering what is actually installed and where.  (I have attached an image of the Start Menu details to better illustrate the issue.)

So my question is, should I uninstall 2.4.1 through the Start Menu? Or have they in fact 'merged' and that Windows (through Start) is simply displaying the structure in an incorrect way?  


![rubyfolder](https://user-images.githubusercontent.com/11639404/30527682-4fb09820-9bf9-11e7-8d7b-b4756e7752e2.png)

**Comment by larskanis** [comment]
I'm glad to hear, that the mintty issue is solved for!

Regarding the start menu entries you're right - this is confusing. And it's actually a flaw in the installer. It doesn't tidy up old entries. I'll fix this in the next release.

You can open a command prompt and type `ridk version` to check your installed ruby version. This is the best way to retrieve version information about the currently running ruby version.

**Comment by larskanis** [comment]
The installer is updated to do uninstall before install when installing a new version of the same Ruby-2.x branch since [this commit](https://github.com/oneclick/rubyinstaller2/commit/f50e16208e5400b4f6b5af1689f969d4c9f907e7). This makes sure, that the entries in the start menu are correct. Uninstalling Ruby will only remove bundled or default gems which comes with RubyInstaller, but will not remove gems installed subsequently.

You can download daily snapshot files of RubyInstaller from the artifacts section from [Appveyor](https://ci.appveyor.com/project/larskanis/rubyinstaller2-hbuor).
